docs/notebooks/synthesize_relational_database.ipynb

```diff
@@ -186,19 +186,25 @@
         "\n",
         "from IPython.display import display, HTML\n",
         "\n",
-        "def join_tables(parent: str, child: str, relational_data=relational_data):\n",
-        "  p_key = relational_data.get_primary_key(parent)\n",
-        "  f_key = \"\"\n",
-        "  for fk in relational_data.get_foreign_keys(child):\n",
-        "    if fk.parent_table_name==parent:\n",
-        "      f_key=fk.column_name\n",
-        "    else:\n",
-        "      logging.warning(\"The input parent and child table must be related.\")\n",
-        "  \n",
-        "  parent_df = relational_data.get_table_data(parent)\n",
-        "  child_df = relational_data.get_table_data(child)\n",
-        "\n",
-        "  joined_data = child_df.merge(parent_df, how=\"left\", left_on=p_key, right_on=f_key)\n",
+        "def join_tables(parent: str, child: str, relational_data: RelationalData, tableset=None):\n",
+        "  keys_between_these_tables = [\n",
+        "    fk for fk in relational_data.get_foreign_keys(child)\n",
```

Comment:
nit: would it be possible to have a check throwing an informative error log message when a child or parent table is passed as argument to the function that are not present in the relational data?

Reply:
Sure, will add

Reply:
I made that + a lack of a relationship between parent and child raise exceptions instead of just logging and leading to some probably-more-cryptic error stacktrace.